lib-dmtf/model/telemetry.go

```diff
@@ -0,0 +1,163 @@
+	TimestampAccuracy       string             `json:"TimestampAccuracy,omitempty"`
+	Units                   string             `json:"Units,omitempty"`
+	Wildcards               []WildCard         `json:"Wildcards,omitempty"`
+}
+
+type CalculationParam struct {
+	ResultMetric string `json:"ResultMetric,omitempty"`
+	SourceMetric string `json:"SourceMetric,omitempty"`
+}
+
+type MetricReportDefinitions struct {
+	ODataID                       string     `json:"@odata.id"`
+	ODataType                     string     `json:"@odata.type"`
+	ID                            string     `json:"Id"`
+	Name                          string     `json:"Name"`
+	AppendLimit                   int        `json:"AppendLimit,omitempty"`
+	Links                         MetricLink `json:"Links,omitempty"`
+	MetricProperties              []string   `json:"MetricProperties,omitempty"`
+	MetricReport                  Oid        `json:"MetricReport,omitempty"`
+	MetricReportDefinitionEnabled bool       `json:"MetricReportDefinitionEnabled,omitempty"`
+	MetricReportDefinitionType    string     `json:"MetricReportDefinitionType,omitempty"`
+	MetricReportHeartbeatInterval string     `json:"MetricReportHeartbeatInterval,omitempty"`
+	Metrics                       []Metric   `json:"Metrics,omitempty"`
+	ReportActions                 []string   `json:"ReportActions,omitempty"`
+	ReportTimespan                string     `json:"ReportTimespan,omitempty"`
+	ReportUpdates                 string     `json:"ReportUpdates,omitempty"`
+	Schedule                      Schedule   `json:"Schedule,omitempty"`
+	Status                        Status     `json:"Status,omitempty"`
+	SuppressRepeatedMetricValue   bool       `json:"SuppressRepeatedMetricValue,omitempty"`
+	Wildcards                     []WildCard `json:"Wildcards,omitempty"`
+}
+
+type Schedule struct {
+	EnabledDaysOfMonth  []string `json:"EnabledDaysOfMonth,omitempty"`
+	EnabledDaysOfWeek   []string `json:"EnabledDaysOfWeek,omitempty"`
+	EnabledIntervals    []string `json:"EnabledIntervals,omitempty"`
+	EnabledMonthsOfYear []string `json:"EnabledMonthsOfYear,omitempty"`
+	InitialStartTime    string   `json:"InitialStartTime,omitempty"`
+	Lifetime            string   `json:"Lifetime,omitempty"`
+	MaxOccurrences      int      `json:"MaxOccurrences,omitempty"`
+	Name                string   `json:"Name,omitempty"`
+	RecurrenceInterval  string   `json:"RecurrenceInterval,omitempty"`
+}
+
+type Metric struct {
+	CollectionDuration  string   `json:"CollectionDuration,omitempty"`
+	CollectionFunction  string   `json:"CollectionFunction,omitempty"`
+	CollectionTimeScope string   `json:"CollectionTimeScope,omitempty"`
+	MetricId            string   `json:"MetricId,omitempty"`
+	MetricProperties    []string `json:"MetricProperties,omitempty"`
+	Oem                 *Oem     `json:"Oem,omitempty"`
+}
+
+type MetricLink struct {
+	Oem      *Oem  `json:"Oem,omitempty"`
+	Triggers []Oid `json:"Triggers,omitempty"`
+}
+
+type Oid struct {
```

Comment:
we should not redfine this. Confirm this is defined elsewhere in the same package and if so remove this.

Reply:
As a separate struct, this is not present.